### PR TITLE
[ACL] Write ACL table/rule creation status into STATE_DB

### PR DESCRIPTION
**Pull request by bingwang-ms**
**What I did**
HLD https://github.com/sonic-net/SONiC/pull/1261
This PR is to update `orchagent` to write ACL table/rule creation status into `STATE_DB`.
Currently, `show acl table` and `show acl rule` commands read ACL table/rule configuration from `CONFIG_DB` directly. We don't know whether the ACL table or rule is created successfully.
We improved `orchagent` to write the status of ACL table/rule into a `STATE_DB` table. 

**Why I did it**
Add the status of ACL table and ACL rule into `STATE_DB` so that user can tell whether the table or rule is created successfully.

**How I verified it**
Verified by copying the updated `orchagent` to a testbed and run.

**Details if related**
HLD https://github.com/sonic-net/SONiC/pull/1261

### CODE REVIEW

**Comment by prsunny** [comment]
You may want to clear the STATE_DB tables during swss restart, otherwise it can show an old data

File: orchagent/aclorch.cpp [hunk 1]
```diff
@@ -4331,34 +4333,48 @@ void AclOrch::doAclTableTask(Consumer &consumer)
...
         }
         else if (op == DEL_COMMAND)
         {
             if (removeAclTable(table_id))
+            {
+                removeAclTableStatus(table_id);
                 it = consumer.m_toSync.erase(it);
+            }
             else
                 it++;
```
[the remark this answers is not in the document]
Comment (bingwang-ms) [reply orchagent/aclorch.cpp:4390]: Yes, I added a `Pending removal` status for this case.

[the remark this answers is not in the document]
Comment (bingwang-ms) [reply orchagent/aclorch.cpp:4390]: Also I added a 'Pending removal` status for the ACL rules that are pending to be removed.

**Comment by bingwang-ms** [comment]
> You may want to clear the STATE_DB tables during swss restart, otherwise it can show an old data

But even if swss restart, the ACL table/rule is still programmed in ASIC, right? So I think we can keep the status in STATE_DB.
There is one scenario I need to handle, that is clear the STATE_DB entries at config_reload. Otherwise there can be stale entries in STATE_DB.

**Comment by bingwang-ms** [comment]
> > You may want to clear the STATE_DB tables during swss restart, otherwise it can show an old data
> 
> But even if swss restart, the ACL table/rule is still programmed in ASIC, right? So I think we can keep the status in STATE_DB. There is one scenario I need to handle, that is clear the STATE_DB entries at config_reload. Otherwise there can be stale entries in STATE_DB.

I added two functions at aclorch startup to clear the status from STATE_DB. This can address both config reload and swss restart scenarios.

**Review by prsunny — APPROVED** [review]

**Pull request merged**